src/components/profile/EditAvatar.js

```diff
@@ -26,10 +27,17 @@ const EditAvatar = ({ screenProps, theme }) => {
   const saveAvatar = async () => {
     setSaving(true)
 
-    await wrappedUserStorage.setProfileField('avatar', avatar, 'public').catch(e => {
-      log.error('saving image failed:', e.message, e)
-      showErrorDialog('We could not capture all your beauty. Please try again.')
-    })
+    await wrappedUserStorage
```

Comment:
1. use async/await dont use then.
2. you can parallelize the saving to storage of avatar and the resizing

Reply:
@sirpy I made it like this for the case if main photo won't be saved then do not start saving small one.

Reply:
still using async await, if await fails it throws an exception and the next line (resizing) wont happen

Reply:
Done. see new commits